thunder/benchmarks/__init__.py

```diff
@@ -3270,6 +3270,161 @@ def fn(self) -> Callable:
+    def description(cls) -> str:
+        return "Benchmark functional optimizers"
+
+    @classmethod
+    @property
+    def args(cls) -> tuple[BenchmarkArg, ...]:
+        return cls._args
+
+    def __init__(
+        self,
+        config: tuple[str, bool, bool | None],
+        params: Sequence[int],
+        device: str = "cuda",
+        dtype: dtypes.dtype = thunder.float32,
+        requires_grad: bool = False,
+        optimizer_name: str = "adam",
+    ) -> None:
+        super().__init__()
+
+        self.config: tuple[str, bool, bool] = config
+        self.params: Sequence[int] = params
+        self.device: str = device
+        self.dtype: dtypes.dtype = dtype
+        self.tdtype: torch.dtype = ltorch.to_torch_dtype(self.dtype)
+        self.requires_grad: bool = requires_grad
+        self.optimizer_name: str = optimizer_name
+
+        self.devices: list[str] = [device]
+
+    def make_batch(self) -> tuple[list, dict]:
+        pt = partial(make_tensor, device=self.device, dtype=self.tdtype, requires_grad=self.requires_grad)
+        params = [pt(shape) for shape in self.params]
+
+        if self.optimizer_name == "adam":
+            grads = [pt(grad) for grad in self.params]
+            exp_avgs = [pt(ea, requires_grad=False) for ea in self.params]
+            exp_avg_sqs = [pt(eas, requires_grad=False) for eas in self.params]
+            max_exp_avg_sqs = [pt(meas, requires_grad=False) for meas in self.params]
+            state_steps = [
+                torch.tensor(0, device=self.device, dtype=self.tdtype, requires_grad=self.requires_grad)
+                for _ in self.params
+            ]
+            return (params, grads, exp_avgs, exp_avg_sqs, max_exp_avg_sqs, state_steps), {}
+
+        elif self.optimizer_name == "sgd":
+            d_p_list = [pt(d_p, requires_grad=False) for d_p in self.params]
+            momentum_buffer_list = [pt(mbl, requires_grad=False) for mbl in self.params]
+            return (params, d_p_list, momentum_buffer_list), {}
+
+        elif self.optimizer_name == "rmsprop":
+            grads = [pt(grad) for grad in self.params]
+            square_avgs = [pt(sq_avgs, requires_grad=False) for sq_avgs in self.params]
+            grad_avgs = [pt(g_avgs, requires_grad=False) for g_avgs in self.params]
+            momentum_buffer_list = [pt(mbl, requires_grad=False) for mbl in self.params]
+            state_steps = [
+                torch.tensor(0, device=self.device, dtype=self.tdtype, requires_grad=self.requires_grad)
+                for _ in self.params
+            ]
+            return (params, grads, square_avgs, grad_avgs, momentum_buffer_list, state_steps), {}
+
+        else:
+            raise ValueError(f"Unsupported optimizer: {self.optimizer_name}")
+
+    def fn(self) -> Callable:
+        def foo(*args):
+
+            if len(self.config) == 2:
+                name, foreach = self.config
+            else:
+                name, foreach, fused = self.config
```

Comment:
```suggestion
            name, foreach, fused = self.config
```

The parametrization has always 3 items in the config tuple, this is not needed, right? 👀

Reply:
I think not always; `rmsprop` currently does not accept the `fused` argument. Ref:

https://github.com/pytorch/pytorch/blob/main/torch/optim/rmsprop.py#L463-L484

Reply:
True, but look here https://github.com/khushi-411/lightning-thunder/blob/287d0aaedfed0d010ab85d12255618996b082a7a/thunder/benchmarks/targets.py#L892-L899 config is always a tuple of length 3